mat/inner.go

```diff
@@ -36,66 +37,84 @@ func Inner(x *VecDense, A Matrix, y *VecDense) float64 {
 			// Panic as a string not a mat.Error.
```

Comment:
I'd rather have `switch a := A.(type) {` and `amat` but it's outside of the diff scope and I think we usually use lower case even for matrix parameters. It's up to you.

Reply:
I'll do that in another PR. Yes, this capital parameter snuck through.